src/tpm/src/std_lib.rs

```diff
@@ -70,21 +72,37 @@ pub unsafe extern "C" fn __fw_malloc(size: usize) -> *mut c_void {
 ///
 /// This function is unsafe because of the parameter of ptr
 pub unsafe extern "C" fn __fw_free(ptr: *mut c_void) {
+    let mut remove = false;
     if let Some(size) = MALLOC_TABLE.lock().get(&(ptr as usize)) {
-        alloc::alloc::dealloc(ptr as *mut u8, Layout::from_size_align_unchecked(*size, 1))
+        alloc::alloc::dealloc(ptr as *mut u8, Layout::from_size_align_unchecked(*size, 1));
+        remove = true;
+    }
+
+    if remove {
+        MALLOC_TABLE.lock().remove_entry(&(ptr as usize));
     }
 }
 
 #[no_mangle]
 /// # Safety
 pub unsafe extern "C" fn __fw_realloc(ptr: *mut c_void, new_size: usize) -> *mut c_void {
+    let mut ptr_new: *mut c_void = null_mut();
```

Comment:
Warning here `value assigned to `ptr_new` is never read`. `ptr_new` is not required to be initialized here?